### DISCUSSION TITLE
[Task]: How to perform training without validation

### DISCUSSION THREAD

**Discussion by Abhijeet99999** [open-edge-platform/anomalib#1609] [Q&A]
Thank you !~

**Comment by willyfh** [comment]
Hello @Abhijeet99999 , by setting the val_split_mode to none, you will prevent Anomalib from validating your model during training. However, even if you don't have any anomaly mask, it is recommended to set val_split_mode to synthetic. This will generate synthetic anomalous images for validation using Perlin noise. By having the validation process, an anomaly score threshold can be adaptively selected, and then used during inference. Similarly, you can also skip the testing phase by setting the test_split_mode to none.

**Comment by Abhijeet99999** [comment]
By Changing code flow, its possible as well. Thank you very much, brother!~

**Comment by Abhijeet99999** [comment]
This will generate synthetic anomalous images for validation using Perlin noise. Where you define perline noise function in the code ?

**Comment by samet-akcay** [comment]
When you set the `val_split_mode` to `synthetic`, the code checks it and generates the synthetic anomalies here
https://github.com/openvinotoolkit/anomalib/blob/470bb860d8478298fdf3f10a530184f8cdc9c938/src/anomalib/data/base/datamodule.py#L156-L159
